Combining PRs #97 & 107: initialize ice fluxes and add "tiice" array

ShanSunNOAA opened the pull request:
Here is a combination of 2 PRs. 

#97: When cplflx=T, initializing ice fluxes by PBL calculated values when fluxes from CICE are unavailable.

#107:
(1) Sfcprop%stc is currently used to store temperatures for both soil and internal ice. In the case of fractional grid, both surface types can exist at one grid point. Thus "tiice" is introduced to store internal ice temperature. "tiice" record will be added to phyf*nc output only when frac_grid=T.
(2) Controling frac_grid in the namelist and no longer modifying it inside FV3GFS_io.F90. frac_grid=F by default.

ShanSunNOAA committed:
(1) Introducing a separate array "tiice" to store internal ice temperature. "tiice" will be added to phyf*nc output only when frac_grid=T.
(2) When cplflx=T, initializing ice fluxes by PBL calculated values when fluxes from CICE are unavailable.
(3) Adding frac_grid to namelist and no longer modifying it inside FV3GFS_io.F90. frac_grid=F by default.

climbfuji commented on gfsphysics/GFS_layer/GFS_diagnostics.F90:
```diff
@@ -2778,6 +2778,22 @@ subroutine GFS_externaldiag_populate (ExtDiag, Model, Statein, Stateout, Sfcprop
+        ExtDiag(idx)%mod_name = 'gfs_sfc'
+        allocate (ExtDiag(idx)%data(nblks))
+        do nb = 1,nblks
+          ExtDiag(idx)%data(nb)%var2 => sfcprop(nb)%tiice(:,num)
```
Use uppercase Sfcprop, please.

climbfuji replied:
Thanks!

climbfuji reviewed (commented):
Where does Sfcprop%tiice get used? I see it is read in and written back to the restart files, but it should get used somewhere in `sfc_sice.f`/`sfc_cice.f`, or?

Also, is the corresponding CCPP PR https://github.com/NCAR/ccpp-physics/pull/429?

Further, I thought we had agreed *not* to accept any further changes to IPD unless they are GFSv16 implementation bugfixes that are brought back to "develop". If we really want to follow this decision, then the changes to `GFS_physics_driver.F90` should be reverted (all others are needed for CCPP as well).

@arunchawla-NOAA @junwang-noaa @DusanJovic-NOAA

SMoorthi-emc commented:
Dom,
  I have to disagree with the IPD decision.
I have some bug fixes to come for both CCPP and IPD.  It does not make
sense to leave bugs in IPD.
I can understand not running regression tests with IPD.
Moorthi

On Wed, May 20, 2020 at 9:03 AM Dom Heinzeller <notifications@github.com>
wrote:

> *@climbfuji* commented on this pull request.
>
> Where does Sfcprop%tiice get used? I see it is read in and written back to
> the restart files, but it should get used somewhere in sfc_sice.f/
> sfc_cice.f, or?
>
> Also, is the corresponding CCPP PR NCAR/ccpp-physics#429
> <https://github.com/NCAR/ccpp-physics/pull/429>?
>
> Further, I thought we had agreed *not* to accept any further changes to
> IPD unless they are GFSv16 implementation bugfixes that are brought back to
> "develop". If we really want to follow this decision, then the changes to
> GFS_physics_driver.F90 should be reverted (all others are needed for CCPP
> as well).
>
> @arunchawla-NOAA <https://github.com/arunchawla-NOAA> @junwang-noaa
> <https://github.com/junwang-noaa> @DusanJovic-NOAA
> <https://github.com/DusanJovic-NOAA>
> ------------------------------
>
> In gfsphysics/GFS_layer/GFS_diagnostics.F90
> <https://github.com/NOAA-EMC/fv3atm/pull/119#discussion_r427988643>:
>
> > @@ -2778,6 +2778,22 @@ subroutine GFS_externaldiag_populate (ExtDiag, Model, Statein, Stateout, Sfcprop
>        ExtDiag(idx)%data(nb)%var2 => Sfcprop(nb)%tsfc(:)
>      enddo
>
> +    if (Model%frac_grid) then
> +      do num = 1,Model%kice
> +        write (xtra,'(i1)') num
> +        idx = idx + 1
> +        ExtDiag(idx)%axes = 2
> +        ExtDiag(idx)%name = 'tiice'//trim(xtra)
> +        ExtDiag(idx)%desc = 'internal ice temperature layer ' // trim(xtra)
> +        ExtDiag(idx)%unit = 'K'
> +        ExtDiag(idx)%mod_name = 'gfs_sfc'
> +        allocate (ExtDiag(idx)%data(nblks))
> +        do nb = 1,nblks
> +          ExtDiag(idx)%data(nb)%var2 => sfcprop(nb)%tiice(:,num)
>
> Use uppercase Sfcprop, please.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/fv3atm/pull/119#pullrequestreview-415305806>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALLVRYSXLIT253DPG2XITX3RSPIK7ANCNFSM4NFSMMDQ>
> .
>


-- 
Dr. Shrinivas Moorthi
Research Meteorologist
Modeling and Data Assimilation Branch
Environmental Modeling Center / National Centers for Environmental
Prediction
5830 University Research Court - (W/NP23), College Park MD 20740 USA
Tel: (301)683-3718

e-mail: Shrinivas.Moorthi@noaa.gov
Phone:  (301) 683-3718   Fax: (301) 683-3718

climbfuji commented:
I am just communicating what was discussed between Arun, Vijay and DTC last week Friday. There was a concern raised that as long as people are able to make changes and develop IPD code, the development will not move over to CCPP entirely.

junwang-noaa commented:
@climbfuji <https://github.com/climbfuji>  I am not sure what is a concern.
ufs-weather-model is running CCPP RT tests, the PR won't be merged without
changes in CCPP.

On Wed, May 20, 2020 at 9:24 AM Dom Heinzeller <notifications@github.com>
wrote:

> I am just communicating what was discussed between Arun, Vijay and DTC
> last week Friday. There was a concern raised that as long as people are
> able to make changes and develop IPD code, the development will not move
> over to CCPP entirely.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/fv3atm/pull/119#issuecomment-631471340>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AI7D6TORZ7BPPQ6L67RQFFTRSPKZDANCNFSM4NFSMMDQ>
> .
>

ShanSunNOAA committed:
Cleaning up.

ShanSunNOAA commented:
Hi Dom,

Yes to all your questions. PR #119 in fv3atm has to be committed together
with PR #429 & #442 in ccpp-physics, where tiice is used in sfc_sice.f. I
should have added this note while creating PR #119. Will do next time.

Thanks for pointing out the inconsistency in PR #97 and #107 in fv3atm,
where the earlier updated IPD and the latter no longer does. I can add
updated sfc_sice.f to IPD for the completeness, if someone points me to a
IPD test to make sure it is working properly, since IPD test is no longer
in the default testing suite, right?

Thanks,
Shan

On Wed, May 20, 2020 at 7:54 AM Jun Wang <notifications@github.com> wrote:

> @climbfuji <https://github.com/climbfuji> I am not sure what is a concern.
> ufs-weather-model is running CCPP RT tests, the PR won't be merged without
> changes in CCPP.
>
> On Wed, May 20, 2020 at 9:24 AM Dom Heinzeller <notifications@github.com>
> wrote:
>
> > I am just communicating what was discussed between Arun, Vijay and DTC
> > last week Friday. There was a concern raised that as long as people are
> > able to make changes and develop IPD code, the development will not move
> > over to CCPP entirely.
> >
> > —
> > You are receiving this because you were mentioned.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/NOAA-EMC/fv3atm/pull/119#issuecomment-631471340>, or
> > unsubscribe
> > <
> https://github.com/notifications/unsubscribe-auth/AI7D6TORZ7BPPQ6L67RQFFTRSPKZDANCNFSM4NFSMMDQ
> >
> > .
> >
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/fv3atm/pull/119#issuecomment-631488518>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALORMVU3TE54Y5ASYOO7SALRSPOJ3ANCNFSM4NFSMMDQ>
> .
>

ShanSunNOAA committed:
Update .gitmodules and submodule pointer for ccpp-physics for code review and testing

climbfuji commented:
> Where does Sfcprop%tiice get used? I see it is read in and written back to the restart files, but it should get used somewhere in `sfc_sice.f`/`sfc_cice.f`, or?
> 
> Also, is the corresponding CCPP PR [NCAR/ccpp-physics#429](https://github.com/NCAR/ccpp-physics/pull/429)?
> 
> Further, I thought we had agreed _not_ to accept any further changes to IPD unless they are GFSv16 implementation bugfixes that are brought back to "develop". If we really want to follow this decision, then the changes to `GFS_physics_driver.F90` should be reverted (all others are needed for CCPP as well).
> 
> @arunchawla-NOAA @junwang-noaa @DusanJovic-NOAA

Most of my questions / concerns have been answered. It is still not clear to me what the exact policy is moving forward with accepting IPD changes. I will note that the changes on the IPD side are partial changes only, the ccpp-physics PR contains more changes (wrt using tiice).

Looking at the CCPP side of things, this PR looks good to me.

ShanSunNOAA committed:
Reverting changes in GFS_physics_driver.F90. These changes are in CCPP code.

ShanSunNOAA commented:
Hi Dom,

Thanks for your prompt reply. Just to confirm, yes, you are right:

(1) ice flux initialization was done earlier, where both IPD and CCPP were
updated.

(2) tiice change was done later and only under CCPP
(GFS_surface_composites.F90 & sfc_sice.f).
Therefore gfsphysics/physics/sfc_sice.f was not updated.

I see your point of being consistent for these 2 PRs, and I have reverted
any changes in GFS_physics_driver.F90.

Let me know if you have any questions! Thanks,

Shan

On Wed, May 20, 2020 at 4:02 PM Dom Heinzeller <notifications@github.com>
wrote:

> Where does Sfcprop%tiice get used? I see it is read in and written back to
> the restart files, but it should get used somewhere in sfc_sice.f/
> sfc_cice.f, or?
>
> Also, is the corresponding CCPP PR NCAR/ccpp-physics#429
> <https://github.com/NCAR/ccpp-physics/pull/429>?
>
> Further, I thought we had agreed *not* to accept any further changes to
> IPD unless they are GFSv16 implementation bugfixes that are brought back to
> "develop". If we really want to follow this decision, then the changes to
> GFS_physics_driver.F90 should be reverted (all others are needed for CCPP
> as well).
>
> @arunchawla-NOAA <https://github.com/arunchawla-NOAA> @junwang-noaa
> <https://github.com/junwang-noaa> @DusanJovic-NOAA
> <https://github.com/DusanJovic-NOAA>
>
> Most of my questions / concerns have been answered. It is still not clear
> to me what the exact policy is moving forward with accepting IPD changes. I
> will note that the changes on the IPD side are partial changes only, the
> ccpp-physics PR contains more changes (wrt using tiice).
>
> Looking at the CCPP side of things, this PR looks good to me.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/fv3atm/pull/119#issuecomment-631751299>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALORMVVRKTMOV4DGIVZ43C3RSRHPNANCNFSM4NFSMMDQ>
> .
>

climbfuji commented:
> Hi Dom, Thanks for your prompt reply. Just to confirm, yes, you are right: (1) ice flux initialization was done earlier, where both IPD and CCPP were updated. (2) tiice change was done later and only under CCPP (GFS_surface_composites.F90 & sfc_sice.f). Therefore gfsphysics/physics/sfc_sice.f was not updated. I see your point of being consistent for these 2 PRs, and I have reverted any changes in GFS_physics_driver.F90. Let me know if you have any questions! Thanks, Shan
> […](#)

Thanks, Shan, consistency is always good.

climbfuji reviewed (approved):
Approved from a CCPP perspective, contingent upon the results of the regression tests.

ShanSunNOAA committed:
Pointing to the latest ccpp/physics branch of add_tiice_20200519

junwang-noaa commented:
> Approved from a CCPP perspective, contingent upon the results of the regression tests.

Dom/Shan,
In CCPP PR #454, for those variables with name changed from "_ocn" to "_wat", can you change the associated comments or the meta data explanation from "open ocean" to "open water" or"over_ocean" to "over_water" in the future commit so that the names and comments/meta data are consistent? Thanks

ShanSunNOAA commented:
Will do. Sorry for missing that.  -Shan

On Tue, May 26, 2020 at 11:40 AM Jun Wang <notifications@github.com> wrote:

> Approved from a CCPP perspective, contingent upon the results of the
> regression tests.
>
> Dom/Shan,
> In CCPP PR #454, for those variables with name changed from "_ocn" to
> "_wat", can you change the associated comments or the meta data explanation
> from "open ocean" to "open water" or"over_ocean" to "over_water" in the
> future commit so that the names and comments/meta data are consistent?
> Thanks
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/fv3atm/pull/119#issuecomment-634171953>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALORMVVLMAL3NSIXT6BHBRDRTP5IPANCNFSM4NFSMMDQ>
> .
>

climbfuji commented:
> > Approved from a CCPP perspective, contingent upon the results of the regression tests.
> 
> Dom/Shan,
> In CCPP PR #454, for those variables with name changed from "_ocn" to "_wat", can you change the associated comments or the meta data explanation from "open ocean" to "open water" or"over_ocean" to "over_water" in the future commit so that the names and comments/meta data are consistent? Thanks

See https://github.com/NCAR/ccpp-physics/issues/455.

ShanSunNOAA committed:
Pointing to the latest ccpp/physics
